File: socket-communication/src/main/java/com/example/socketcommunication/global/error/ErrorResponse.java
package com.example.socketcommunication.global.error;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ErrorResponse {

    private final int status;

    private final String message;

    public String convertObjectToJson(Object object) throws JsonProcessingException {
        return new ObjectMapper().writeValueAsString(object);
    }
}

File: practice-kotlin/src/main/kotlin/com/example/practicekotlin/global/error/ErrorResponse.kt
package com.example.practicekotlin.global.error

import com.example.practicekotlin.global.error.exception.BusinessException

class ErrorResponse (
    val status: Int,
    val message: String
) {

    companion object {
        fun of(e: BusinessException): ErrorResponse {
            return ErrorResponse(
                status = e.status,
                message = e.message
            )
        }
    }

}